ats/parser/parser.py

```diff
@@ -77,12 +77,16 @@ def process_program():
 
     # procedure : ‘procedure’ proc_name ‘{‘ stmtLst ‘}’
     def process_procedure():
+        nonlocal current_token
         match_token("procedure")
         name = match_name_token()
         match_token("{")
         node = process_stmt_lst()
         match_token("}")
 
+        if current_token == "procedure":
+            process_procedure()
+
```

Comment:
@Monikaa1909, @patrycjarecko: to powinno byc w `process_program`

@zwolkowy: ast